src/settings/persistent_settings_handler.cpp

```diff
@@ -143,11 +143,10 @@ void mp::PersistentSettingsHandler::set(const QString& key, const QString& val)
 std::set<QString> mp::PersistentSettingsHandler::keys() const
 {
     std::set<QString> ret{};
-    std::transform(
-        cbegin(settings),
-        cend(settings),
-        std::inserter(ret, begin(ret)),
-        [](const auto& elem) { return elem.first; }); // I wish get<0> worked here... maybe in C++20
+    std::transform(cbegin(settings),
+                   cend(settings),
+                   std::inserter(ret, begin(ret)),
+                   [](const auto& elem) { return elem.first; });
```

Comment:
`std::get<0>` still refers to a set of overloads, so there's not (yet) a way to pass that overload-set as an argument.

Reply:
This is a good idea, but it'd probably make sense to go through more of our codebase to find places we could adopt ranges. I'll file an issue for that so that we can keep track of it.

Reply:
https://github.com/canonical/multipass/issues/4412